client/components/breadcrumb/index.tsx

```diff
@@ -41,33 +54,38 @@ interface Props {
 const Breadcrumb: React.FunctionComponent< Props > = ( { items, compact = false } ) => {
 	const translate = useTranslate();
 
-	return (
-		<StyledUl>
-			{ compact && items.length > 1 ? (
-				<StyledLi>
-					<StyledGridicon icon="chevron-left" size={ 18 } />
-					{ /*  Show the exactly previous page with items[ items.length - 2 ] */ }
-					<a href={ items[ items.length - 2 ].href }>{ translate( 'Back' ) }</a>
-				</StyledLi>
-			) : (
-				<>
-					{ items.map( ( item: { href?: string; label: string }, index: Key ) => {
-						return (
-							<StyledLi key={ index }>
-								{ compact && <StyledGridicon icon="chevron-left" size={ 18 } /> }
-								{ index !== 0 && <StyledGridicon icon="chevron-right" size={ 18 } /> }
-								{ item.href ? (
-									<a href={ item.href }>{ item.label }</a>
-								) : (
-									<span>{ item.label }</span>
-								) }
-							</StyledLi>
-						);
-					} ) }
-				</>
-			) }
-		</StyledUl>
-	);
+	if ( compact && items.length > 1 ) {
+		return (
+			<StyledBackLink href={ items[ items.length - 2 ].href }>
+				<Gridicon icon="chevron-left" size={ 18 } />
+				{ /*  Show the exactly previous page with items[ items.length - 2 ] */ }
+				{ translate( 'Back' ) }
+			</StyledBackLink>
+		);
+	}
+
+	if ( items.length > 1 ) {
+		return (
+			<StyledUl>
+				{ items.map( ( item: { href?: string; label: string }, index: Key ) => (
+					<StyledLi key={ index }>
+						{ index !== 0 && <StyledGridicon icon="chevron-right" size={ 14 } /> }
+						{ item.href && index !== items.length - 1 ? (
+							<a href={ item.href }>{ item.label }</a>
+						) : (
+							<span>{ item.label }</span>
+						) }
+					</StyledLi>
+				) ) }
+			</StyledUl>
+		);
+	}
+
+	if ( items.length === 1 ) {
+		return <StyledRootLabel>{ items[ 0 ].label }</StyledRootLabel>;
+	}
+
+	return null;
```

Comment:
is this really needed? would it make sense moving this top the top in case `items.length === 0`?

Reply:
Yes, it is needed because otherwise, it would return `undefined` (if `length` is empty) and the current version of React does not support it. As a side note, latest React 18 allows that as [explained here](https://github.com/reactwg/react-18/discussions/75).

I will move it to the top and make it more explicit as you suggest

Reply:
Update on the comment above. Typescript enforces to return at least a value, otherwise, it raises the following warning
```
Type '{ (props: PropsWithChildren<Props>): Element | null | undefined; defaultProps: Partial<Props> | undefined; }' is not assignable to type 'FunctionComponent<Props>'.
  Type 'Element | null | undefined' is not assignable to type 'ReactElement<any, any> | null'.
    Type 'undefined' is not assignable to type 'ReactElement<any, any> | null'.
```
which is a way to say `Undefined cannot be returned`